### DISCUSSION TITLE
How i can access to the parent component of the component that i'm dropping in the editor?

### DISCUSSION THREAD

**Discussion by mohamad97-alhomsi** [GrapesJS/grapesjs#4599] [Q&A] [answered]
I'm trying to limit the component's children for specific components, for example, I have a component and need to put just one child inside it so how i can achieve this?
**Note**: I've already solved this for the component that exists in the main layer [in the Editor] but what I need to do if that component is positioned  inside many parents

**Comment by artf** [comment, marked as answer]
Hi @mohamad97-alhomsi you should be able to configure your component with [properties](https://grapesjs.com/docs/api/component.html#properties) like `draggable`/`droppable`

**Reply by mohamad97-alhomsi** [reply]
Yes, I know that 
but I think you misunderstood my question
**For example**, I have a component named Text that it's a container [not a text type just an empty container named text]
so I want to drag and drop this text component into any other component, which means that component would be a child for many parents 
Now once I drop any other component inside that Text component we're talking about I need to prevent dropping any second component inside that text which means i need this text component to have just one child.
is that possible to do?
thanks.

**Reply by artf** [reply]
>is that possible to do? 

Yes, by using what I've suggested:
```js
// for your Text component
droppable: (src, trg) => trg.components().length < 2
```

**Reply by mohamad97-alhomsi** [reply]
Thanks a lot, 😀
it worked
